### ISSUE TITLE
addon gateways appear DOA

### ISSUE THREAD

**Issue by sdake** [istio/istio#12301]
**Describe the bug**
certificate error when accessing addon IPs.

<img width="829" alt="screen shot 2019-03-06 at 1 10 31 pm" src="https://user-images.githubusercontent.com/755849/53910584-67b44580-4011-11e9-8174-e9ad34eb02ec.png">

**Expected behavior**
access to the https enabled endpoint

**Steps to reproduce the bug**
```sdake@beast-01:~/repro$ more istio-deploy.sh
#!/bin/sh

helm install istio/install/kubernetes/helm/istio-init --name istio-init --namespace istio-system
# May neeed to increase this on slower cloud providers.
sleep 10
helm upgrade --install istio istio/install/kubernetes/helm/istio --namespace istio-system --timeout=600 --set global.tag=sdake --set global.hub=docker.io/sdake --set tracing.enabled=true --set grafana.enabled=true --set prometheus.enabled=true --set tracing.gateway.enabled=true --set grafana.gateway.enabled=true --set tracing.gateway.enabled=true --set prometheus.gateway.enabled=true
```

**Version**
istio 1.1.0 release

**Installation**
see reeproducer

**Environment**
kubeadm deployed cluster, kubernetes 1.14.4, metallb 0.7.3

**Comment by douglas-reid** [comment]
I've recently encountered this as well. We should either move this back to being HTTP or fix the certs.

**Comment by liminw** [comment]
@wattli @myidpt @incfly

**Comment by incfly** [comment]
@JimmyCYJ

**Comment by jmazzitelli** [comment]
Do we know when/how this started breaking? I'm curious what changed to cause this to start failing.

**Comment by duderino** [comment]
We discussed this today.  Notes:

This is a P0 doc issue for @philrud to document how to use letsencrypt certs for the addon gateways.   Phil already has documentation for using the certs, but any doc that describes the addon gateways must describe/reference how to use the letsencrypt certs

There is also a P0 task for @douglas-reid to revert the new gateways.yaml from the addon directories

**Comment by myidpt** [comment]
Thanks for the update, @duderino

**Comment by sdake** [comment]
@jmazzitelli the history is this:
1. addons were HTTP only and could be exposed only via port-forward
1. we added gateways to expose the HTTP endpoint in the ingressgateway
1. the HTTP endpoint was determined as insecure for prod, so HTTPS endpoints were added
1. The implementation of HTTPS endpoints didn't properly expose certs.
1. The PR at the end of all this activity indicated work left undone relating to certs.  Unfortunately that work was missed until the last few days, and the consensus was that the risk of implementing correct certs was too great to achieve in a few days of activity.

@duderino has explained well the corrective action we will take in this case.  This is a P0 documentation bug at this point and IMO must be documented as this likely blocks many production users of the addons.
